src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -816,77 +799,80 @@ void YlmSpherepack::fill_vector_work_arrays() const noexcept {
 
   // Initialize vector work arrays
   const size_t ldwg = (3 * n_theta_ * (n_theta_ + 3) + 2) / 2;
-  {
-    auto& dworkg = memory_pool_.get(ldwg);
-    int err = 0;
-    vhsgsi_(static_cast<int>(n_theta_), static_cast<int>(n_phi_),
-            work_vector_spec_to_phys.data(), static_cast<int>(l_vhsgs),
-            dworkg.data(), static_cast<int>(ldwg), &err);
-    memory_pool_.free(dworkg);
-    ASSERT(0 == err, "Error in vhsgsi");
+  auto& dworkg = memory_pool_.get(ldwg);
+  int err = 0;
+  vhsgsi_(static_cast<int>(n_theta_), static_cast<int>(n_phi_),
+          work_vector_spec_to_phys.data(), static_cast<int>(l_vhsgs),
+          dworkg.data(), static_cast<int>(ldwg), &err);
+  memory_pool_.free(dworkg);
+  if (err != 0) {
+    ERROR("vhsgsi error " << err << " in YlmSpherepack");
   }
 }
 
 void YlmSpherepack::fill_scalar_work_arrays() const noexcept {
   auto& work_phys_to_spec = storage_.work_phys_to_spec;
-  auto& work_spec_to_phys = storage_.work_spec_to_phys;
+  auto& work_scalar_spec_to_phys = storage_.work_scalar_spec_to_phys;
   auto& quadrature_weights = storage_.quadrature_weights;
 
-  std::vector<double> weights(n_theta_);
   {
+    auto& weights = memory_pool_.get(n_theta_);
     auto& work0 = memory_pool_.get(n_theta_);
     auto& work1 = memory_pool_.get(n_theta_ + 1);
     int err = 0;
     gaqd_(static_cast<int>(n_theta_), work0.data(), weights.data(),
           work1.data(), static_cast<int>(n_theta_ + 1), &err);
     memory_pool_.free(work0);
     memory_pool_.free(work1);
-    ASSERT(0 == err, "Failure in gaqd");
-  }
+    if (err != 0) {
```

Comment:
we should annotate all of the error checks with `if (UNLIKELY(err != 0))` to tell the compiler that we do not expect this to evaluate true very frequently (in practice more than once per simulation per core). The macro is defined in `Utilities/Gsl.hpp`